devito/function.py

```diff
@@ -348,14 +349,26 @@ def argument_values(self, alias=None, **kwargs):
         # Add value override for own data if it is provided
         if self.name in kwargs:
             new = kwargs.pop(self.name)
+            if isinstance(new, TimeFunction):
+                if new.save != self.save:
+                    raise TypeError("Operator generated for save=%s only," %
+                                    self.save +
```

Comment:
Use template strings, not ``+`` to concatenate them (much clearer; also, this convention is loosely followed throughout Devito)